### PR TITLE
github workflow for tekton

### PR DESCRIPTION
**Pull request by joejstuart**

### CODE REVIEW

File: .github/workflows/tekton.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: setup kind cluster
+        uses: container-tools/kind-action@v2
+
+      - name: Run ec task
+        run: |
```
Comment (lcarva) [inline .github/workflows/tekton.yaml:40]: Can we toss all of this into a bash script so we can run it against a local kind cluster?

Comment (joejstuart) [reply .github/workflows/tekton.yaml:40]: Absolutely, I still have some cleanup to do.

File: deployments/ecp-policy.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+kind: EnterpriseContractPolicy
+metadata:
+  creationTimestamp: null
+  name: ec-policy
+spec:
+  description: My custom enterprise contract policy configuration
+  sources:
+    - git::https://github.com/hacbs-contract/ec-policies.git
+  exceptions:
+    nonBlocking:
```
Comment (zregvart) [inline deployments/ecp-policy.yaml:11]: For followup: we should be using the non-deprecated `configuration/excludeRules` here

File: .github/workflows/tekton.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+          # install tekton and wait for controllers to start
+          tekton_version=v0.38.4
+          kubectl apply --filename https://storage.googleapis.com/tekton-releases/pipeline/previous/$tekton_version/release.yaml
+          kubectl wait pods -n tekton-pipelines -l pipeline.tekton.dev/release=$tekton_version --for=condition=Ready --timeout=90s
+
+          # setup resources for the ec task to run
+          kubectl create secret generic cosign-public-key --from-file=cosign.pub=hack/cosign.pub
+          kustomize build https://github.com/hacbs-contract/enterprise-contract-controller/config/crd?ref=main | kubectl apply -f -
```
Comment (zregvart) [inline .github/workflows/tekton.yaml:48]: I prefer using `kubectl kustomize`, one less binary to make sure is available...

File: .github/workflows/tekton.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+      - name: setup kind cluster
+        uses: container-tools/kind-action@v2
+
+      - name: Run ec task
+        run: |
+          # install tekton and wait for controllers to start
+          tekton_version=v0.38.4
+          kubectl apply --filename https://storage.googleapis.com/tekton-releases/pipeline/previous/$tekton_version/release.yaml
+          kubectl wait pods -n tekton-pipelines -l pipeline.tekton.dev/release=$tekton_version --for=condition=Ready --timeout=90s
```
Comment (zregvart) [inline .github/workflows/tekton.yaml:44]: We could probably wait more towards the end and proceed with the creation of other resources; probably doesn't make much of a difference...

File: deployments/verify-enterprise-contract-v2-taskrun.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+spec:
+  params:
+  - name: HOMEDIR
+    value: /tekton/home
+  - name: IMAGES
+    value: |
+      {
+        "components": [
+          {
+            "containerImage": "quay.io/hacbs-contract-demo/single-container-app@sha256:84550a596e40024e6ea10ce5aab9ad5b0b91a14cb27161654722ab1d11c70ac2"
```
Comment (zregvart) [inline deployments/verify-enterprise-contract-v2-taskrun.yaml:25]: Would be cool to pull this from `hack/images.txt` so we don't need to think about updating it here when that changes

File: deployments/verify-enterprise-contract-v2-taskrun.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+          }
+        ]
+      }
+  - name: POLICY_CONFIGURATION
+    value: $(context.taskRun.namespace)/ec-policy
+  - name: PUBLIC_KEY
+    value: k8s://$(context.taskRun.namespace)/cosign-public-key
+  - name: REKOR_HOST
+    value: ""
```
Comment (zregvart) [inline deployments/verify-enterprise-contract-v2-taskrun.yaml:34]: This is the default, so just wondering if we need it here?

**Review by zregvart — APPROVED** [review]
Very nice! In terms of followup on this in a separate PR, we could install everything via kustomize; i.e. all those separate kubectl apply/create could be merged into a single `kustomization.yaml` file.

Another followup could be to get this workflow triggered via `repository_dispatch` on changes to ec-policies repository.

I think it is more beneficial to get this merged and then modify as needed.

File: .github/workflows/tekton.yaml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: setup kind cluster
+        uses: container-tools/kind-action@v2
+
+      - name: Run ec task
+        run: |
+          # install tekton and wait for controllers to start
+          tekton_version=v0.38.4
```
[the remark this answers is not in the document]
Comment (joejstuart) [reply .github/workflows/tekton.yaml:42]: I might have to come back to this. v0.38.4 is the latest version that works with the version of kubernetes that's being used.

**Pull request merged**